projects/plugins/social/src/class-note.php

```diff
@@ -11,10 +11,45 @@
  * Register the Jetpack Social Note custom post type.
  */
 class Note {
+	const JETPACK_SOCIAL_NOTE_CPT                      = 'jetpack-social-note';
+	const JETPACK_SOCIAL_REWRITE_RULES_LAST_FLUSHED_AT = 'jetpack_socia_rewrite_rules_last_flushed_at';
+
+	/**
+	 * Check if the feature is enabled.
+	 */
+	public function enabled() {
+		return defined( 'JETPACK_SOCIAL_NOTES_ENABLED' ) && constant( 'JETPACK_SOCIAL_NOTES_ENABLED' );
+	}
+
+	/**
+	 * Initialize the Jetpack Social Note custom post type.
+	 */
+	public function init() {
+		flush_rewrite_rules();
```

Comment:
I think we might want to call this with the `false` parameter to make it a soft flush https://developer.wordpress.org/reference/functions/flush_rewrite_rules/